### ISSUE TITLE
API for internally calculated fields

### ISSUE THREAD

**Issue by tpapp** [JuliaObjects/ConstructionBase.jl#57]
I read the docs but it is still not clear how one should define the relevant interface for types that have internally calculated fields. Eg
```julia
struct Foo
    a
    b
    c
    Foo(a, b) = new(a, b, hypot(a, b))
end
```
I want the user to be able to modify (= create a new immutable with) `a` and `b`, and pretend that `c` does not even exist.

**Comment by jw3126** [comment]
I agree this is a pain point. What I generally recommend is using `setproperties`/`getproperties` for public access and `constructorof` for private access. See also https://github.com/JuliaObjects/ConstructionBase.jl/pull/54#issuecomment-1088765017

**Comment by jw3126** [comment]
Did you try to overload `ConstructionBase.setproperties`?
